### PR TITLE
fix: end the FileStreamWriter for the installer file

### PR DESCRIPTION
**Pull request by colemickens**
##### Description

We have a failure on a repo where `Error: spawn ETXTBSY` was observed after #46 was merged.

Taking a closer look,  we should be ~~`close()`-ing the WritableStream, and then~~ `end()`ing the FileStreamWriter.

ref:
* ~~`WritableStream`: `close()`: https://nodejs.org/api/webstreams.html#writablestreamclose~~
* `FileStreamWriter`: `end()`: https://nodejs.org/api/stream.html#writableendchunk-encoding-callback

##### Checklist

- [ ] Tested changes against a test repository
- [ ] Added or updated relevant documentation (leave unchecked if not applicable)
- [ ] (If this PR is for a release) Updated README to point to the new tag (leave unchecked if not applicable)

### CODE REVIEW

**Comment by colemickens** [comment]
Update:

* `pipeTo()` closes the WritableStream itself (ref: https://2ality.com/2022/06/web-streams-nodejs.html ctrl-f "writablestream is closed")

Random:

* this seems useful for preventing mistakes of not awaiting awaitables: https://typescript-eslint.io/rules/no-floating-promises/

File: src/main.ts [hunk 1]
```diff
@@ -437,6 +437,7 @@ class NixInstallerAction {
         const fileStream = fs.createWriteStream(tempfile);
         const fileStreamWeb = stream.Writable.toWeb(fileStream);
         await response.body.pipeTo(fileStreamWeb);
+        fileStream.end();
```
Comment (Hoverbear) [inline src/main.ts:440]: Referring to https://nodejs.org/api/fs.html#filehandlecreatewritestreamoptions, I'm not sure this syncs the file. I wonder if we also need to call...
https://nodejs.org/api/fs.html#filehandlesync

Comment (colemickens) [reply src/main.ts]: what interesting is that the code I changed wasn't calling sync explicitly, and was just waiting for the finished signal. (which I'd assume is similar to awaiting the `pipeTo`)

I'm going to going to really wack this with a big hammer, open the file so I get a filehandle, make sure it gets fsync'd explicitly and see how this fares.

Comment (colemickens) [reply src/main.ts]: further musing: there's a "synchronous append" mode (`as`) for opening, but not an equivalent for `w`? But there is `rs+`:

> 'rs+': Open file for reading and writing in synchronous mode. Instructs the operating system to bypass the local file system cache.

**Comment by Hoverbear** [comment]
Unfortunately this results in an error when running the checks:

```bash
nix-installer-action on  colemickens/fix-stream-close [$!] is 📦 v1.0.0 via  v20.8.0 via ❄️  impure (nix-shell-env) took 3s 
❯ npm run all

> nix-installer-action@1.0.0 all
> npm run build && npm run format && npm run lint && npm run package


> nix-installer-action@1.0.0 build
> tsc

src/main.ts:12:25 - error TS2307: Cannot find module 'fetch-retry' or its corresponding type declarations.

12 import fetchRetry_ from "fetch-retry";
                           ~~~~~~~~~~~~~


Found 1 error in src/main.ts:12

```

I realize now we don't actually note that we use the same style as https://github.com/actions/javascript-action so you need to run `npm run all` before every commit. :( We need to add this to the readme.

**Comment by Hoverbear** [comment]
#48 should fix the lack of a check!

**Comment by colemickens** [comment]
I agree we need #48 (and that I had a diff in `dist`), but the error you got was from not re-running `npm install`, it definitely in the `package-lock.json` at the most recent commit.

**Comment by colemickens** [comment]
I rebased this on #48, hopefully we can merge it first if we want it, otherwise I'll rebase this again.

**Comment by colemickens** [comment]
now I'm hitting: https://github.com/nodejs/node/issues/48466

**Review by Hoverbear — APPROVED** [review]

**Pull request merged**